### PR TITLE
Enable k32l2b

### PR DESCRIPTION
**Pull request by jhqian**
Add board support for frdm_k32l2b3

Add board specific pinmux, doc and dts.
Tested with hello_world, synchronization, philosophers, basic/blinky and basic/button. 

Depends on zephyrproject-rtos/hal_nxp#22

### CODE REVIEW

File: boards/arm/frdm_k32l2b3/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,230 @@
+.. _frdm_k32l2b3:
+
+NXP FRDM-K32L2B3
+##############
```
Comment (dbkinder) [inline boards/arm/frdm_k32l2b3/doc/index.rst]: Title underlines need to be at least as long as the title above it, so add a couple more ``#``s

File: boards/arm/frdm_k32l2b3/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+the `OpenSDA DAPLink FRDM-K32L2B3 Firmware`_.
+
+Option 2: :ref:`opensda-jlink-onboard-debug-probe`
+--------------------------------------------------
+
+Install the :ref:`jlink-debug-host-tools` and make sure they are in your search
+path.
+
+Follow the instructions in :ref:`opensda-jlink-onboard-debug-probe` to program
+the `OpenSDA J-Link FRDM-K32L2B3 Firmware`_.
```
Comment (dbkinder) [inline boards/arm/frdm_k32l2b3/doc/index.rst]: I see you've got a "OpenSDA J-Link Generic Firmware" link defined at the bottom of the doc, so this should probably refer to that?

File: boards/arm/frdm_k32l2b3/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+The FRDM-K32L2B3 Freedom development board provides a platform for evaluation
+and development of the K32 L2B MCU Family. The board includes onboard debug
+probe, segment LCD, accelerometer/magnetometer, a full speed USB and easy
+access to K32 L2B's MCU I/O. The FRDM-K32L2B3 board is fully supported by the
+MCUXpresso suite of tools, which provides device drivers, middleware and
+examples to allow rapid development, plus configuration tools and an optional
+free IDE.
+
+.. image:: ./frdm_k32l2b3.jpg
```
Comment (dbkinder) [inline boards/arm/frdm_k32l2b3/doc/index.rst]: the filename uploaded has a dash in it: ``frdm-k32l2b3.jpg`` so rename the image filename or fix this name to match

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/frdm_k32l2b3/doc/index.rst (hunk 3 above)
Comment (jhqian) [reply boards/arm/frdm_k32l2b3/doc/index.rst]: Thank you, @dbkinder . just updated the doc accordingly.

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks.

File: modules/Kconfig.mcux [hunk 4]
```diff
@@ -6,7 +6,7 @@
 config HAS_MCUX
 	bool
 	select HAS_CMSIS_CORE
-	depends on SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || SOC_FAMILY_LPC
+	depends on SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || SOC_FAMILY_LPC || SOC_FAMILY_K32L2
```
Comment (MaureenHelm) [inline modules/Kconfig.mcux]: This should be a new series under the Kinetis family, not a new family.

File: boards/arm/frdm_k32l2b3/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+Build and flash applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Configuring a Debug Probe
+=========================
+
+A debug probe is used for both flashing and debugging the board. This board is
+configured by default to use the :ref:`opensda-daplink-onboard-debug-probe`.
+
+Early versions of this board have an outdated version of the OpenSDA bootloader
```
Comment (MaureenHelm) [inline boards/arm/frdm_k32l2b3/doc/index.rst]: Is this true for a new board?

File: boards/arm/frdm_k32l2b3/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+and require an update. Please see the `DAPLink Bootloader Update`_ page for
+instructions to update from the CMSIS-DAP bootloader to the DAPLink bootloader.
+
+Option 1: :ref:`opensda-daplink-onboard-debug-probe` (Recommended)
+------------------------------------------------------------------
+
+Install the :ref:`pyocd-debug-host-tools` and make sure they are in your search
+path.
```
Comment (MaureenHelm) [inline boards/arm/frdm_k32l2b3/doc/index.rst]: pyocd doesn't support this soc yet, does it?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by hakehuang** [comment]
@jhqian , I find there is a problem with samples/subsys/usb/cdc_acm with the -DOVERLAY_CONFIG=overlay-composite-cdc-msc.conf, it reports 
```

[ 93%] Built target drivers__serial

Scanning dependencies of target zephyr_prebuilt

[ 94%] Building C object zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj

[ 95%] Linking C executable zephyr_prebuilt.elf

Memory region         Used Size  Region Size  %age Used

           FLASH:       54804 B       256 KB     20.91%

            SRAM:       32996 B        32 KB    100.70%

        IDT_LIST:          72 B         2 KB      3.52/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: zephyr_prebuilt.elf section `usb_descriptor' will not fit in region `SRAM'

/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: section .intList VMA [0000000020006000,0000000020006047] overlaps section usb_descriptor VMA [0000000020005fcc,0000000020006099]

/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: region `SRAM' overflowed by 228 bytes

collect2: error: ld returned 1 exit status

%

zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:97: recipe for target 'zephyr/zephyr_prebuilt.elf' failed

make[2]: *** [zephyr/zephyr_prebuilt.elf] Error 1

CMakeFiles/Makefile2:662: recipe for target 'zephyr/CMakeFiles/zephyr_prebuilt.dir/all' failed

make[1]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/all] Error 2

Makefile:83: recipe for target 'all' failed

make: *** [all] Error 2

script returned exit code 2

```

**Review by hakehuang — APPROVED** [review]
except one usb sample code issue, there is no other exist application issue on this porting.

File: boards/arm/frdm_k32l2b3/doc/index.rst (hunk 6 above)
Comment (jhqian) [reply boards/arm/frdm_k32l2b3/doc/index.rst]: remove that statement for now, in fact, I'm working on a PR

File: boards/arm/frdm_k32l2b3/doc/index.rst (hunk 5 above)
Comment (jhqian) [reply boards/arm/frdm_k32l2b3/doc/index.rst]: removed

**Comment by jhqian** [comment]
Not sure what happened on the CI system side, the failed tickless_concept can't be reproduced on my side.

**Comment by jhqian** [comment]
Not sure about the error reported in during shippable checking.

**Pull request closed without merging**